File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020307Activity.java
package com.liushu.crazyandroid.ui.stage01.chapter02.activity;

import android.os.SystemClock;
import android.view.View;
import android.widget.Button;
import android.widget.Chronometer;
import android.widget.ImageView;
import android.widget.TextView;

import com.jaydenxiao.common.base.BaseActivity;
import com.liushu.crazyandroid.R;

import butterknife.BindView;
import butterknife.OnClick;

public class Demo020307Activity extends BaseActivity {
    @BindView(R.id.iv_back)
    ImageView mIvBack;
    @BindView(R.id.tv_title_name)
    TextView mTvTitleName;
    @BindView(R.id.chronometer_test)
    Chronometer mChronometerTest;
    @BindView(R.id.btn_test)
    Button mBtnTest;

    @Override
    public int getLayoutId() {
        return R.layout.activity_demo020307;
    }


    @Override
    public void initView() {
        mTvTitleName.setText("手机里的“劳力士”");
        mBtnTest.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mChronometerTest.setBase(SystemClock.elapsedRealtime());
                mChronometerTest.start();
                mBtnTest.setClickable(false);
            }
        });
        mChronometerTest.setOnChronometerTickListener(new Chronometer.OnChronometerTickListener() {
            @Override
            public void onChronometerTick(Chronometer chronometer) {
                if (SystemClock.elapsedRealtime() - chronometer.getBase() > 20 * 1000) {
                    mChronometerTest.stop();
                    mBtnTest.setClickable(true);
                }
            }
        });
    }

    @OnClick(R.id.iv_back)
    public void onClick() {
        finish();
    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020307Activity.kt
package com.liushu.crazyandroid.ui.stage01.chapter02.activity

import android.os.SystemClock
import android.widget.Chronometer
import butterknife.OnClick
import com.jaydenxiao.common.base.BaseActivity
import com.liushu.crazyandroid.R
import kotlinx.android.synthetic.main.activity_demo020307.*
import kotlinx.android.synthetic.main.head_title_layout.*

class Demo020307Activity : BaseActivity() {

    override val layoutId: Int
        get() = R.layout.activity_demo020307


    override fun initView() {
        tv_title_name!!.text = "手机里的“劳力士”"
        btn_test!!.setOnClickListener {
            chronometer_test!!.base = SystemClock.elapsedRealtime()
            chronometer_test!!.start()
            btn_test!!.isClickable = false
        }
        chronometer_test!!.onChronometerTickListener = Chronometer.OnChronometerTickListener { chronometer ->
            if (SystemClock.elapsedRealtime() - chronometer.base > 20 * 1000) {
                chronometer_test!!.stop()
                btn_test!!.isClickable = true
            }
        }
    }

    @OnClick(R.id.iv_back)
    fun onClick() {
        finish()
    }
}
